profile/src/main/java/com/scottlogic/deg/profile/validators/profile/constraints/atomic/TemporalConstraintValidator.java

```diff
@@ -19,20 +39,35 @@ public TemporalConstraintValidator(String rule, List<FieldDTO> fields)
     public final ValidationResult validate(TemporalConstraintDTO dto)
     {
         ValidationResult fieldMustBeValid = fieldMustBeValid(dto);
-        if(!fieldMustBeValid.isSuccess) return fieldMustBeValid;
+        if (!fieldMustBeValid.isSuccess) return fieldMustBeValid;
 
-        ValidationResult dateMustBeSpecified = dateMustBeSpecified(dto);
-        if(!dateMustBeSpecified.isSuccess) return dateMustBeSpecified;
+        ValidationResult dateTimeMustBeValid = dateTimeMustBeValid(dto);
+        if (!dateTimeMustBeValid.isSuccess) return dateTimeMustBeValid;
 
-        return valueMustBeValid(dto, FieldType.DATETIME);
+        return fieldTypeMustMatchValueType(dto, FieldType.DATETIME);
     }
 
-
-    private ValidationResult dateMustBeSpecified(TemporalConstraintDTO dto)
+    private ValidationResult dateTimeMustBeValid(TemporalConstraintDTO dto)
     {
-        String date = dto.getDate();
-        return date != null && !date.isEmpty()
-            ? ValidationResult.success()
-            : ValidationResult.failure("Date must be specified" + getErrorInfo(dto));
+        String dateTime = dto.getDate();
+        if (dateTime == null || dateTime.isEmpty())
+        {
+            return ValidationResult.failure("DateTime must be specified" + getErrorInfo(dto));
+        }
+        try
+        {
+            OffsetDateTime offsetDateTime = DateTimeFactory.create(dateTime);
+            OffsetDateTime max = DateTimeDefaults.get().max();
+            OffsetDateTime min = DateTimeDefaults.get().min();
+            if (offsetDateTime.compareTo(max) > 0 || offsetDateTime.compareTo(min) < 0)
+            {
+                return ValidationResult.failure("Dates must be between 0001-01-01T00:00:00.000Z and 9999-12-31T23:59:59.999Z" + getErrorInfo(dto));
+            }
+            return ValidationResult.success();
+        }
+        catch (Exception e)
```

Comment:
Can we catch a more specific exception here?

Reply:
Maybe, but as we will do the same logic regardless (printing the message), I don't think it would make any difference.